src/languages/smtlib/lexSmtlib.mll

```diff
@@ -42,35 +42,53 @@ rule token = parse
   | [' ' '\t' '\r']+    { token lexbuf }
   | '\n'                { Lexing.new_line lexbuf; token lexbuf }
   | ';' (_ # '\n')*     { token lexbuf }
+
   | '_'                 { UNDERSCORE }
+  | '!'                 { ATTRIBUTE }
   | "as"                { AS }
   | "let"               { LET }
-  | "forall"            { FORALL }
   | "exists"            { EXISTS }
-  | '!'                 { ATTRIBUTE }
-  | "par"               { raise Error }
+  | "forall"            { FORALL }
+  | "match"             { raise Error }
+  | "par"               { PAR }
+
   | "NUMERAL"           { raise Error }
   | "DECIMAL"           { raise Error }
   | "STRING"            { raise Error }
-  | "set-logic"         { SET_LOGIC }
-  | "set-option"        { SET_OPTION }
-  | "set-info"          { SET_INFO }
-  | "declare-sort"      { DECLARE_SORT }
-  | "define-sort"       { DEFINE_SORT }
-  | "declare-fun"       { DECLARE_FUN }
-  | "define-fun"        { DEFINE_FUN }
-  | "push"              { PUSH }
-  | "pop"               { POP }
+
+  | "not"               { NOT }
+
   | "assert"            { ASSERT }
+  | "check-sat-assuming"        { raise Error }
```

Comment:
Could you add the corresponding token (and associated syntax rules), rather thna raising Error ? (Not absolutely necessary as I can do it after merging the PR)